src/FeetMinimumJerkGenerator.cpp

```diff
@@ -0,0 +1,339 @@
+/*
+ * Copyright (C) 2018 Fondazione Istituto Italiano di Tecnologia
+ * Authors: Giulio Romualdi, Stefano Dafarra
+ * CopyPolicy: Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
+ *
+ */
+
+#include <FeetMinimumJerkGenerator.h>
+#include <iDynTree/Core/VectorDynSize.h>
+#include <iDynTree/Core/CubicSpline.h>
+#include <iDynTree/Core/EigenHelpers.h>
+#include "iDynTree/Core/Twist.h"
+#include <cassert>
+#include <mutex>
+
+class FeetMinimumJerkGenerator::FeetMinimumJerkGeneratorImplementation {
+public:
+
+    double stepHeight = -1.0, pitchDelta = 0.0, swingApex = 0.5, landingVelocity = 0.0;
+
+    iDynTree::VectorDynSize xPositionsBuffer, yPositionsBuffer, zPositionsBuffer, yawsBuffer, pitchAnglesBuffer, timesBuffer, zTimesBuffer;
+
+    std::vector<iDynTree::Transform> leftTrajectory, rightTrajectory;
+    std::vector<iDynTree::Twist> leftMixedTwist, rightMixedTwist;
+
+    std::mutex mutex;
+
+    FeetMinimumJerkGeneratorImplementation()
+        : xPositionsBuffer(2)
+        , yPositionsBuffer(2)
+        , zPositionsBuffer(3)
+        , yawsBuffer(2)
+        , pitchAnglesBuffer(3)
+        , timesBuffer(2)
+        , zTimesBuffer(3)
+    { }
+
+    bool minimumJerk(const double &initialPoint,
+                     const double &finalPoint,
+                     const double &t,
+                     const double &trajectoryDuration,
+                     double &positionOutput,
+                     double &velocityOutput,
+                     double &accelerationOutput)
+    {
+        double normalizedTime = t/trajectoryDuration;
+        if((normalizedTime < 0) || (normalizedTime > 1)){
+            std::cerr << "[FeetMinimumJerkGenerator::minimumJerk] The time has to be a positive number less than 1 "
+                      << " Remember to normalize the time."
+                      << std::endl;
+            return false;
+        }
+
+        double p = 6 * std::pow(normalizedTime,5) - 15 * std::pow(normalizedTime,4) + 10 * std::pow(normalizedTime,3);
+        double dp = (5 * 6 * std::pow(normalizedTime,4) - 4 * 15 * std::pow(normalizedTime,3) + 3 * 10 * std::pow(normalizedTime,2)) / trajectoryDuration;
+        double ddp = (4 * 5 * 6 * std::pow(normalizedTime,3) - 3 * 4 * 15 * std::pow(normalizedTime,2) + 2 * 3 * 10 * normalizedTime) / std::pow(trajectoryDuration, 2);
+        positionOutput = p * (finalPoint - initialPoint) + initialPoint;
+        velocityOutput = dp * (finalPoint - initialPoint);
+        accelerationOutput = ddp * (finalPoint - initialPoint);
+        return true;
+    }
+
+
+    bool interpolateFoot(double dT, const std::vector<size_t>& phaseShift, const std::vector<StepPhase> &stepPhase,
+                         const FootPrint &foot, std::vector<iDynTree::Transform> &output, std::vector<iDynTree::Twist> &outputTwistsInMixedRepresentation) {
```

Comment:
What is missing is the relation between the RPY double derivative and the angular acceleration. From the outside it is not clear that inside the RPY parametrization is used, thus I would avoid to output the RPY double derivative instead of the actual acceleration. We may address this in a separate PR.